### PR TITLE
dapp-docs: added source code documentation (#438)

### LINKED ISSUE #438: [System Contract DApp] Phase 4: Source code documentation
### Problem

The project necessitates comprehensive source code documentation to serve as a guide for onboarding individuals to the project. This documentation should be thorough and meticulous in describing the purposes of each component within the project. Additionally, the document should include a tutorial section that provides guidance on how to integrate new system contracts into the DApp.

### Solution

- add a new documentation to the dapp

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by quiet-node**
**Description**: This PR added a new source code document which is thorough and meticulous in describing the purposes of each component within the project. Additionally, the document should include a tutorial section that provides guidance on how to integrate new system contracts into the DApp which can be found in the README.md of the source-code-document folder

**Related issue(s)**: #438 

Fixes #438 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by AlfredoG87 — APPROVED** [review]
LGTM

**Review by Nana-EC — APPROVED** [review]
LG
Note the spelling issue with the directory Fredy noted.

File: system-contract-dapp-playground/package.json [hunk 1]
```diff
@@ -17,7 +17,7 @@
     "build": "next build",
     "start": "next start",
     "lint": "next lint",
-    "prerequisite-check": "ts-node ./scripts/prerequisite-check.ts",
+    "prerequisite-check": "ts-node ./preprequisite-check/scripts/index.ts",
```
[the remark this answers is not in the document]
Comment (quiet-node) [reply system-contract-dapp-playground/package.json]: it's indeed a typo! Thanks for pointing it out Fredy!

**Review by AlfredoG87 — APPROVED** [review]
LG 👍

**Pull request merged**